src/wazuh_db/wdb_integrity.c

```diff
@@ -241,98 +242,101 @@ static void wdbi_update_completion(wdb_t * wdb, wdb_component_t component, long
 
     sqlite3_bind_int64(stmt, 1, timestamp);
     sqlite3_bind_int64(stmt, 2, timestamp);
-    sqlite3_bind_text(stmt, 3, COMPONENT_NAMES[component], -1, NULL);
+    sqlite3_bind_text(stmt, 3, manager_checksum, -1, NULL);
+    sqlite3_bind_text(stmt, 4, COMPONENT_NAMES[component], -1, NULL);
 
     if (sqlite3_step(stmt) != SQLITE_DONE) {
         mdebug1("DB(%s) sqlite3_step(): %s", wdb->id, sqlite3_errmsg(wdb->db));
     }
 }
 
 // Query the checksum of a data range
-int wdbi_query_checksum(wdb_t * wdb, wdb_component_t component, const char * command, const char * payload) {
-    int retval = -1;
-    cJSON * data = cJSON_Parse(payload);
+integrity_sync_status_t wdbi_query_checksum(wdb_t * wdb, wdb_component_t component, dbsync_msg action, const char * payload) {
+    integrity_sync_status_t status = INTEGRITY_SYNC_ERR;
 
+    // Parse payload
+    cJSON * data = cJSON_Parse(payload);
     if (data == NULL) {
         mdebug1("DB(%s): cannot parse checksum range payload: '%s'", wdb->id, payload);
         return -1;
     }
-
     cJSON * item = cJSON_GetObjectItem(data, "begin");
     char * begin = cJSON_GetStringValue(item);
-
     if (begin == NULL) {
         mdebug1("No such string 'begin' in JSON payload.");
         goto end;
     }
-
     item = cJSON_GetObjectItem(data, "end");
     char * end = cJSON_GetStringValue(item);
-
     if (end == NULL) {
         mdebug1("No such string 'end' in JSON payload.");
         goto end;
     }
-
     item = cJSON_GetObjectItem(data, "checksum");
     char * checksum = cJSON_GetStringValue(item);
-
     if (checksum == NULL) {
         mdebug1("No such string 'checksum' in JSON payload.");
         goto end;
     }
-
     item = cJSON_GetObjectItem(data, "id");
-
     if (!cJSON_IsNumber(item)) {
         mdebug1("No such string 'id' in JSON payload.");
         goto end;
     }
-
     long timestamp = item->valuedouble;
-    os_sha1 hexdigest;
-    struct timespec ts_start, ts_end;
-    gettime(&ts_start);
 
-    switch (wdbi_checksum_range(wdb, component, begin, end, hexdigest)) {
-    case -1:
-        goto end;
+    os_sha1 manager_checksum = {0};
+    // Get the previously computed manager checksum
+    if (INTEGRITY_CHECK_GLOBAL == action) {
+        if (OS_SUCCESS == wdbi_get_last_manager_checksum(wdb, component, manager_checksum) && 0 == strcmp(manager_checksum, checksum)) {
+            status = INTEGRITY_SYNC_CKS_OK;
+        }
+    }
 
-    case 0:
-        retval = 0;
-        break;
+    // Get the actual manager checksum
+    if (status != INTEGRITY_SYNC_CKS_OK) {
+        struct timespec ts_start, ts_end;
+        gettime(&ts_start);
+        switch (wdbi_checksum_range(wdb, component, begin, end, manager_checksum)) {
```

Comment:
We are calculating the checksum with `wdbi_checksum_range`() even if the action is **INTEGRITY_CHECK_LEFT**, but this action doesn't seem to need it.

Reply:
**wdbi_checksum_range()** is necessary even when the action is **INTEGRITY_CHECK_LEFT** because every action must return **status** with the result of the checksum comparison.